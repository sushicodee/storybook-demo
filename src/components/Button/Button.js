import React from 'react';
import "./Button.scss";

const Button = ({label,handleSubmit,input}) => {

    return (
        <div  className = "todo-button">
            <button data-testid = "button" disabled = {!input} onClick = {e => handleSubmit(e)}>{label}</button>
        </div>
    )
}
export default Button
