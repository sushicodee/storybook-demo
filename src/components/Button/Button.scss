.todo-button{
    button{
        margin-top:10px;
        padding:10px 5px;
        margin-left:10px;
        border-radius: 4px;
        border:2px solid rgb(11, 201, 11);
        color:rgb(11, 201, 11);

        &:enabled {
            color:rgba(255, 255, 255, 0.774);
            background-color: rgb(29, 228, 29);
        }
        &:hover {
            color:white;
            background-color: rgb(0, 230, 38);
        }
    }
}