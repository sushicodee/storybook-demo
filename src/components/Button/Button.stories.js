import React from "react";
import { storiesOf ,action} from "@storybook/react";
import Button from "./Button";

const defaultProps = {
    label :"create",
    handleSubmit:() => action('handleSubmit'),
    input:''
}

const populatedInputProps = {
    label :"create",
    handleSubmit:() =>action('handleSubmit'),
    input:"Buy Milk"
}

storiesOf("TodoButton",module)
.add("default", () => <Button {...defaultProps}/>)
.add("whenInput", () => <Button {...populatedInputProps}/>)