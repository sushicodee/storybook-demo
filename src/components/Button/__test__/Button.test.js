import Button from './../Button';
import ReactDom from 'react-dom';
import React from 'react';
import {render,cleanUp} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'

// afterEach(cleanUp);

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDom.render(<Button/> , div);
})

it('renders button correctly', () => {
    const props = {
        label:"Clear Todos"
    }
     const {getByTestId} = render(<Button label = {props.label}/>)
     expect(getByTestId('button'))
     .toHaveTextContent(props.label);
})

it('renders button correctly', () => {
    const props = {
        label:"create"
    }
     const {getByTestId} = render(<Button label = {props.label}/>)
     expect(getByTestId('button'))
     .toHaveTextContent(props.label);
})

// it('mock implementation of basic function', () => {
//     const mock = jest.fn(() => 'i am a mock function');
//     expect(mock('Calling my mock function').toBe('i am a mock function'));
//     expect(mock).toHaveBeenCalledWith('Calling my mock function');
// })

it('matches snapshot' , () => {
    const props = {
        label:'create Todo'
    }
    const tree = renderer.create(<Button label = {props.label}/>).toJSON();
    expect(tree).toMatchSnapshot();
})