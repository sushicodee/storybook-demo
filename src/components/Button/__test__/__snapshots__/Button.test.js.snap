// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="todo-button"
>
  <button
    data-testid="button"
    disabled={true}
    onClick={[Function]}
  >
    create Todo
  </button>
</div>
`;
